lib/cloudflare_protected_email_address.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'cgi'
+
+class CloudflareProtectedEmailAddress
+  def initialize(obfuscated)
+    @obfuscated = obfuscated
+  end
+
+  def human_readable
+    return CGI.unescape(escaped_email_address) if CGI.unescape(escaped_email_address).include?('@')
+    obfuscated
```

Comment:
This has become much too complicated and confusing now, I fear. If you really need to do this check, I'd suggest at least factoring out that `CGI.unescape(escaped_email_address)` to a named method.